demo/site/preBuild/src/createRedirects.ts

```diff
@@ -47,9 +47,23 @@ const createApiRedirects = async (): Promise<Redirect[]> => {
     for (const redirect of response.redirects) {
         let source: string | undefined;
         let destination: string | undefined;
+        let has: Redirect["has"];
 
         if (redirect.sourceType === "path") {
-            source = redirect.source;
+            source = redirect.source.replace(/:/g, "\\:"); // escape colon, otherwise it is used for next.js regex path matching  (https://nextjs.org/docs/pages/api-reference/next-config-js/redirects#regex-path-matching)
+
+            // query parameters have to be defined with has, see: https://nextjs.org/docs/pages/api-reference/next-config-js/redirects#header-cookie-and-query-matching
+            if (source?.includes("?")) {
+                const searchParams = new URLSearchParams(source.split("?")[1]);
+                source = source.split("?")[0];
+                has = [];
+
+                searchParams.forEach((value, key) => {
+                    if (has) {
```

Comment:
I suppose the if is unnecessary since you initialize the variable in line 59

Reply:
unfortunately not, it would work like this:

<img width="469" alt="Screenshot 2023-08-30 at 07 56 10" src="https://github.com/vivid-planet/comet/assets/44967610/5c5e644b-6f12-4e70-b8e3-f5a9e55a7b3e">

but as soon as I'm using forEach it throws an error: 
<img width="385" alt="Screenshot 2023-08-30 at 07 56 50" src="https://github.com/vivid-planet/comet/assets/44967610/3573b218-c337-4a76-9708-ffd2a74dcea2">


(but I can't use entries() since I'm getting an error with our current ts config)
<img width="860" alt="Screenshot 2023-08-30 at 07 59 06" src="https://github.com/vivid-planet/comet/assets/44967610/8caa619b-8f43-4736-b0e0-2cdea6ef8890">

Reply:
Ah okay 👍